VideoWeb/VideoWeb/ClientApp/src/app/app.component.ts

```diff
@@ -92,17 +92,15 @@ export class AppComponent implements OnInit, OnDestroy {
 
     private async postAuthSetup(loggedIn: boolean) {
         this.loggedIn = loggedIn;
-        await this.attemptRetrieveProfile(loggedIn);
+        await this.attemptRetrieveProfile();
         this.checkBrowser();
         this.setPageTitle();
         this.setupSubscribers();
         this.connectionStatusService.start();
     }
 
-    private async attemptRetrieveProfile(loggedIn: boolean) {
-        if (!loggedIn && !this.isSignInUrl) {
-            this.router.navigate([`/${pageUrls.IdpSelection}`]);
-        } else {
+    private async attemptRetrieveProfile() {
+        if (this.loggedIn || this.isSignInUrl) {
             await this.retrieveProfileRole();
         }
```

Comment:
Could just be in postAuthSetup() instead of having another intermediary function.